x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlDailyMaintenanceService.java

```diff
@@ -70,9 +72,11 @@ public MlDailyMaintenanceService(ClusterName clusterName, ThreadPool threadPool,
     private static TimeValue delayToNextTime(ClusterName clusterName) {
         Random random = new Random(clusterName.hashCode());
         int minutesOffset = random.ints(0, MAX_TIME_OFFSET_MINUTES).findFirst().getAsInt();
-        DateTime now = DateTime.now(ISOChronology.getInstance());
-        DateTime next = now.plusDays(1).withTimeAtStartOfDay().plusMinutes(30).plusMinutes(minutesOffset);
-        return TimeValue.timeValueMillis(next.getMillis() - now.getMillis());
+
+        Instant instant = ZonedDateTime.now().toInstant();
+        ZonedDateTime now = IsoChronology.INSTANCE.zonedDateTime(instant, ZoneId.systemDefault());
```

Comment:
how about `Instant.now().toEpochMilli()?`

Reply:
actually - this one is better to use ZonedDateNow as it can use java.time dsl to calculate a `next` variable 
`next = now.plusDays(1).toLocalDate().atStartOfDay(now.getZone()).plusMinutes(30).plusMinutes(minutesOffset);`

Reply:
actually I ZonedDatetime, might better here - more on a new review https://github.com/elastic/elasticsearch/pull/35949/files#r236714576